test/Microsoft.AspNetCore.Sockets.Tests/ConnectionManagerTests.cs

```diff
@@ -183,9 +185,49 @@ public void ScanAfterDisposeNoops()
             connectionManager.Scan();
         }
 
-        private static ConnectionManager CreateConnectionManager()
+        [Fact]
+        public async Task ApplicationLifetimeIsHookedUp()
+        {
+            var appLifetime = new TestApplicationLifetime();
+            var connectionManager = CreateConnectionManager(appLifetime);
+
+            appLifetime.Start();
+
+            var connection = connectionManager.CreateConnection();
+
+            appLifetime.StopApplication();
+
+            // Connection should be disposed so this should complete immediately
+            Assert.False(await connection.Application.Out.WaitToWriteAsync().OrTimeout());
+        }
+
+        private static ConnectionManager CreateConnectionManager(IApplicationLifetime lifetime = null)
         {
-            return new ConnectionManager(new Logger<ConnectionManager>(new LoggerFactory()));
+            lifetime = lifetime ?? new TestApplicationLifetime();
```

Comment:
Looks like the `EmptyApplicationLifetime` should be the default - if nothing is passed...